src/ILCompiler.ReadyToRun/src/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -491,5 +492,42 @@ public ISymbolNode ComputeConstantLookup(ReadyToRunHelperId helperId, object ent
         {
             return _compilation.NodeFactory.ReadyToRunHelper(helperId, entity, token);
         }
+
+        private void setVars(CORINFO_METHOD_STRUCT_* ftn, uint cVars, NativeVarInfo* vars)
+        {
+            Debug.Assert(_debugVarInfos == null);
+
+            if (cVars == 0)
+                return;
+
+            _debugVarInfos = new DebugVarInfo[1];
+            _debugVarInfos[0] = new DebugVarInfo();
+            _debugVarInfos[0].Ranges = new List<NativeVarInfo>();
+            for (int i = 0; i < cVars; i++)
+            {
+                _debugVarInfos[0].Ranges.Add(vars[i]);
+            }
+        }
+
+        // Create a DebugLocInfo which is a table from native offset to source line.
+        // using native to il offset (pMap) and il to source line (_sequencePoints).
+        private void setBoundaries(CORINFO_METHOD_STRUCT_* ftn, uint cMap, OffsetMapping* pMap)
+        {
+            Debug.Assert(_debugLocInfos == null);
+            
+            List<DebugLocInfo> debugLocInfos = new List<DebugLocInfo>();
+            for (int i = 0; i < cMap; i++)
+            {
+                OffsetMapping nativeToILInfo = pMap[i];
+                NativeLocInfo nativeLocInfo = new NativeLocInfo { nativeOffset = nativeToILInfo.nativeOffset, ilOffset = nativeToILInfo.ilOffset, source = (uint)nativeToILInfo.source };
+                DebugLocInfo loc = new DebugLocInfo(0, null, nativeLocInfo, 0);
+                debugLocInfos.Add(loc);
+            }
+
+            if (debugLocInfos.Count > 0)
+            {
+                _debugLocInfos = debugLocInfos.ToArray();
```

Comment:
Why do we have to add everything to List<DebugLocInfo> and then convert it to an array? It seems that we know the number of entries upfront (cMap), or did I miss something?

Reply:
Fixed. Yes we can just new it up upfront. This code was simplified from the Jit version that required a double pass.